test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/ODataDynamicPropertyTests.cs

```diff
@@ -487,6 +487,18 @@ public void SerializationIgnoresContainerPropertyIfEdmTypeIsNotOpen()
             VerifyMessageBody(expectedResult, messageBody);
         }
 
+        [Fact]
+        public void SerializationWithNullValueAsDynamicProperty()
+        {
+            var director = new Director { Id = 1, Name = "Director 1" };
+            director.DynamicProperties.Add("Title", null);
+
+            var messageBody = SerializeEntity(this.directorEntitySetName, director);
+            var expectedResult = "{\"@odata.type\":\"#ServiceNS.Director\",\"Id\":1,\"Name\":\"Director 1\"}";

```

Comment:
> var expectedResult = "{\"@odata.type\":\"#ServiceNS.Director\",\"Id\":1,\"Name\":\"Director 1\"}"; [](start = 11, length = 99)

do we support to upload a dynamic property with null value in client?

Reply:
@xuzhg Based on the spec, [a missing dynamic property is defined to be the same as a dynamic property with value null](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_SetaValuetoNull). I therefore reasoned there's no reason to send dynamic properties with null values over the wire. We don't serialize dynamic properties with null values in OData WebAPI either. This test verifies that if we come across such a property in the dynamic properties container property, we just skip it.

Reply:
For example:  if we want to remove a dynamic property value as null for a certain entity, we should "patch" that entity with "null" dynamic property. How can we do this in Client?

Reply:
That would be achieved by using a declared property on the client side. This way the type annotation will be written in the payload for that dynamic property. Plus it'd make it easy to do property tracking for that property.

Given that the container property is a dictionary of `string` and `object`, we wouldn't know the type annotation to write if we came across `KeyValuePair<string, object>` with a `null` value during serialization. We'd have no problem serializing it as `"propertyName":null`, but it'd be minus type annotation. I'm not confident the service would handle it gracefully with type info missing. Which is why I don't think the container property is a good fit for this scenario.